libs/application/templates/directorate-of-immigration/citizenship/src/components/DescriptionText.tsx

```diff
@@ -0,0 +1,101 @@
+import { ReactNode } from 'react'
+import Markdown from 'markdown-to-jsx'
+import { MessageDescriptor, useIntl } from 'react-intl'
+import {
+  Box,
+  Text,
+  BulletList,
+  Bullet,
+  Link,
+  TextProps,
+} from '@island.is/island-ui/core'
+import * as styles from './DescriptionText.css'
+
+const BulletListBox = ({ children }: { children: ReactNode }) => {
+  return (
+    <Box marginBottom={3}>
+      <BulletList space={2}>{children}</BulletList>
+    </Box>
+  )
+}
+
+const TextComponent = ({ children, ...props }: { children: ReactNode }) => {
+  return (
+    <Box className={styles.paragraphContainer}>
+      <Text {...props}>{children}</Text>
+    </Box>
+  )
+}
+
+const LinkComponent = ({
+  children,
+  href,
+}: {
+  children: ReactNode
+  href: string
+}) => {
+  return (
+    <Link href={href} className={styles.link}>
+      {children}
+    </Link>
+  )
+}
+
+interface Props {
+  text: MessageDescriptor
+  format?: { [key: string]: string | number }
+  textProps?: TextProps
+}
+
+const headingOverride = {
+  component: Text,
+  props: {
+    variant: 'h4',
+    marginBottom: 1,
+  },
+}
+
+const textOverrideProps: TextProps = {
+  marginBottom: 2,
+}
+
+const DescriptionText = ({ text, format, textProps }: Props) => {
+  const { formatMessage } = useIntl()
+  const markdown = formatMessage(text, format)
+  // markdown-to-jsx is able to handle this in most cases but when using 'formatMessage'
+  // it does not work for some reason. That is the reason for this special handling here.
+  // We will take a look at this later with the localization team.
```

Comment:
I am using using this to format text components inside react components, not in the form building.
Like in fields/StaysAbroad/StaysAbroadRepeaterItem.tsx , I need to format the text with an integer variable and am using this there, is there a way around this?

Reply:
Ah gotcha, there is a workaround that would remove the need of the DescriptionText
You could replace
```diff
- <DescriptionText
-   text={information.labels.staysAbroad.itemTitle}
-   format={{ index: itemNumber + 1 }}
-   textProps={{
-     as: 'h5',
-     fontWeight: 'semiBold',
-     marginBottom: 0,
-   }}
- />

+ {DescriptionFormField({
+   application: application,
+   showFieldName: false,
+   field: {
+     id: 'title',
+     title: '',
+     description: formatMessage(
+       information.labels.staysAbroad.itemTitle,
+       { index: itemNumber + 1 }
+     ),
+     titleVariant: 'h5',
+     type: FieldTypes.DESCRIPTION,
+     component: FieldComponents.DESCRIPTION,
+     children: undefined,
+   },
+ })}

And then change the message string to 
itemTitle: {
        id: 'doi.cs.application:information.labels.staysAbroad.itemTitle',
+       defaultMessage: '**Dvalarland {index}**',
        description: 'Stay abroad item separator title',
      },

Reply:
one question here, if the string comes from contentful, which it will eventually, will it be bold? since we are doing the ** to bold it in the defaultMessage

Reply:
Ah right, so for the markdown messages you need to add #markdown at the end of the id in order for contentful to grab that its a markdown field. You then get the rich text editor on contentful for that string.
So this string should be
id: 'doi.cs.application:information.labels.staysAbroad.itemTitle#markdown',